tests/functional/test_project_quota.py

```diff
@@ -0,0 +1,108 @@
+# pylint: disable=missing-module-docstring,redefined-outer-name
+import pytest
+
+from .conftest import oc
+
+
+# LKS: Cannot differentiate between "project does not exist" and
+# "operation failed for some other reason".
+def test_project_quota_project_notfound(session):
+    """Test that a request for the quota of a project that does not exist fails
+    as expected"""
+    res = session.get("/projects/does-not-exist/quota")
+    assert res.status_code == 400
+
+
+def test_project_quota_no_quota(session, a_project):
+    """Test that a request for the quota of a project that has no quota
+    returns a null quota"""
+
+    res = session.get(f"/projects/{a_project}/quota")
+    assert res.status_code == 200
+    data = res.json()
+    assert data["Kind"] == "MocQuota"
+    assert all(data["Quota"][k] is None for k in data["Quota"])
+
+
+@pytest.mark.parametrize("multiplier", [0, 1, 2])
+def test_project_quota_set_quota(session, a_project, multiplier):
+    """Test that setting a quota with different multipliers results in
+    the expected quota values"""
+
+    expected = {
+        "configmaps": "4",
+        "limits.ephemeral-storage": f"{2 + (8 * multiplier)}Gi",
+        "requests.ephemeral-storage": f"{2 + (8 * multiplier)}Gi",
+        "openshift.io/imagestreams": "2",
+        "persistentvolumeclaims": "2",
+        "replicationcontrollers": "2",
+        "requests.storage": "2",
+        "resourcequotas": "5",
+        "secrets": "4",
+        "services": "4",
+        "services.loadbalancers": "2",
+        "services.nodeports": "2",
+    }
+    res = session.put(
```

Comment:
this depends on the quota definition that is passed in (and probably modified by an operator).  I considered testing it this way, but opted to test it a different way that I put in my pull request.

Reply:
That's correct; this assumes you're test environment has loaded the quotas from `k8s/base/quotas.json`, which seems reasonable. We could make the test conditional, but I think it's important to test this behavior.

Reply:
I agree that this is important, however, every time the configuration file is changed, you need to modify this test.  I would prefer to work out a way to test this without having to modify the test code every time we change the configuration file.  

I approached this was to consider all of the quotas as linear functions, by finding the slope based on what is returned from varying the multiplier.  This way the tests could be run in a production environment without specific knowledge of the quotas were configured.  using oc, we could always get the config map and compare the values from the api to confirm that it is working.

Reply:
Having to modify the unit test when the sample quota definition file is updated is a reasonable assumption. Static files are easier to debug than logic code.

Reply:
I'm fine with this, however, I think in the future we should consider ways that we could test this in production to confirm that a particular installation is working.